src/cow-react/modules/limitOrders/pure/ReceiptModal/FilledField.tsx

```diff
@@ -11,31 +15,116 @@ interface Props {
 }
 
 export function FilledField({ order, sellAmount, buyAmount }: Props) {
-  const { inputToken, outputToken } = order
+  const {
+    inputToken,
+    outputToken,
+    filledPercentage,
+    fullyFilled,
+    kind,
+    sellToken,
+    buyToken,
+    feeAmount,
+    executedBuyAmount,
+    executedSellAmount,
+    executedFeeAmount,
+    filledAmount,
+  } = order
 
-  const filledPercentage = useMemo(() => {
-    if (!order || !order.filledPercentage) {
+  const touched = !!filledPercentage?.gt(0)
+
+  let mainToken: Token
+  let mainAddress: string
+  let mainAmount: CurrencyAmount<Token>
+  let swappedToken: Token
+  let swappedAddress: string
+  let swappedAmount: JSBI | undefined
+  let action: string
+
+  let filledAmountWithFee, swappedAmountWithFee
+  if (kind === OrderKind.SELL) {
+    action = 'sold'
+
+    mainToken = inputToken
+    mainAddress = sellToken
+    mainAmount = sellAmount.add(CurrencyAmount.fromRawAmount(mainToken, feeAmount.toString()))
+
+    swappedToken = outputToken
+    swappedAddress = buyToken
+    swappedAmount = executedBuyAmount
+
+    // Sell orders, add the fee in to the sellAmount (mainAmount, in this case)
+    filledAmountWithFee = filledAmount?.plus(executedFeeAmount?.toString() || '0')
+    swappedAmountWithFee = new BigNumber(swappedAmount?.toString() || '0')
+  } else {
+    action = 'bought'
+
+    mainToken = outputToken
+    mainAddress = buyToken
+    mainAmount = buyAmount
+
+    swappedToken = inputToken
+    swappedAddress = sellToken
+    swappedAmount = executedSellAmount
+
+    // Buy orders need to add the fee, to the sellToken too (swappedAmount in this case)
+    filledAmountWithFee = filledAmount
+    swappedAmountWithFee = new BigNumber(swappedAmount?.toString() || '0').plus(executedFeeAmount || '0')
+  }
```

Comment:
it feels like all of the above could be moved to a function to keep the render function simpler and easier to reason about